compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -967,14 +969,44 @@ pub fn expand_format_args_nl<'cx>(
     expand_format_args_impl(ecx, sp, tts, true)
 }
 
+fn emit_errors_for_unused_named_argument_names(
+    names: FxHashMap<Symbol, (usize, Span)>,
+    cx: &Context<'_, '_>,
+    unverified_pieces: Vec<parse::Piece<'_>>,
+) {
+    let mut used_argument_names = FxHashSet::<&str>::default();
+    for piece in unverified_pieces {
+        if let rustc_parse_format::Piece::NextArgument(a) = piece {
+            match a.position {
+                rustc_parse_format::Position::ArgumentNamed(arg_name, _) => {
+                    used_argument_names.insert(arg_name);
+                }
+                _ => {}
+            };
+            match a.format {
+                FormatSpec { width: Count::CountIsName(s, _), .. } => {
+                    used_argument_names.insert(s);
+                }
+                _ => {}
```

Comment:
Oh, it seems like you were doing that already? 🤔

Reply:
It turns out I missed the same case but for `precision`!